controllers/templatesync/template_sync.go

```diff
@@ -1161,24 +1155,57 @@ func (r *PolicyReconciler) processDependencies(
 			continue
 		}
 
-		// set up namespace for replicated policy dependencies
-		ns := dep.Namespace
-		if ns == "" {
-			ns = r.ClusterNamespace
-		}
+		var res dynamic.ResourceInterface
 
-		// query object and compare compliance status to desired
-		res := dClient.Resource(rsrc).Namespace(ns)
+		if namespaced {
+			ns := dep.Namespace
+			if ns == "" && dep.Group == policiesv1.GroupVersion.Group {
+				// ocm policies should always be in the cluster namespace
+				ns = r.ClusterNamespace
+			}
+
+			res = dClient.Resource(rsrc).Namespace(ns)
+		} else {
+			res = dClient.Resource(rsrc)
+		}
 
 		depObj, err := res.Get(ctx, dep.Name, metav1.GetOptions{})
 		if err != nil {
+			if dep.Group == utils.GvkConstraintTemplate.Group && templateDeps[dep] != "Compliant" {
```

Comment:
Do you need to check that the error is that it wasn't found/CRD not installed?

Reply:
That kind of context could be nice, I'll see what I can do.

Reply:
I think I misunderstood what you were actually asking, and started implementing a way to help users understand reasons for a dependency not being satisfied... which is still good, and helpful! But I think you were actually just requesting that this case be checked to ensure that the object was really "not found" as opposed to some other error?

Reply:
That's right. It looks like you fixed that in the latest push.